ocaml-lsp-server/src/hover_req.mli

```diff
@@ -0,0 +1,3 @@
+open Import
+
+val run : State.t Server.t -> State.t -> HoverParams.t -> Hover.t option Fiber.t
```

Comment:
This needs a module doc strings.

Reply:
Do we need to pass the state? Isn't it a part of the server already.

Reply:
Yeah, but now we've given this function an explicit interface. It's important to keep that tidy. Whatever `later` does is less important. And `later` should probably stop passing the state altogether. It's confusing to be able to acquire it from two places.

Reply:
Do you propose 
option a) ignore the state passed within `later` callback and get state from the server 
option b) first send another PR where `later` doesn't pass `state` in the callback at all ? 

I'm not very confident in my understanding of how `later` works, so option b will take a couple of days. It would be nice to remove #561's blocker faster.

Reply:
a) is fine. So is b)

If this refactoring work for another PR you could always just include it in that PR as separate commits. What if we decide to reject the other PR? Or what if the requirements for that PR change?